core/capabilities/transmission/transmission.go

```diff
@@ -0,0 +1,108 @@
+package transmission
+
+import (
+	"fmt"
+	"time"
+
+	"golang.org/x/crypto/sha3"
+
+	"github.com/smartcontractkit/libocr/permutation"
+	ragep2ptypes "github.com/smartcontractkit/libocr/ragep2p/types"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+	p2ptypes "github.com/smartcontractkit/chainlink/v2/core/services/p2p/types"
+)
+
+// TODO determine location for this code
+
+var (
+	// S = [N]
+	Schedule_AllAtOnce = "allAtOnce"
+	// S = [1 * N]
+	Schedule_OneAtATime = "oneAtATime"
+)
+
+type TransmissionConfig struct {
+	Schedule   string
+	DeltaStage time.Duration
+}
+
+func ExtractTransmissionConfig(config *values.Map) (TransmissionConfig, error) {
+	var tc struct {
+		DeltaStage string
+		Schedule   string
+	}
+	err := config.UnwrapTo(&tc)
+	if err != nil {
+		return TransmissionConfig{}, fmt.Errorf("failed to unwrap tranmission config from value map: %w", err)
+	}
+
+	duration, err := time.ParseDuration(tc.DeltaStage)
+	if err != nil {
+		return TransmissionConfig{}, fmt.Errorf("failed to parse DeltaStage %s as duration: %w", tc.DeltaStage, err)
+	}
+
+	return TransmissionConfig{
+		Schedule:   tc.Schedule,
+		DeltaStage: duration,
+	}, nil
+}
+
+// GetPeerIDToTransmissionDelay returns a map of PeerID to the time.Duration that the node with that PeerID should wait
+// before transmitting. If a node is not in the map, it should not transmit.
+func GetPeerIDToTransmissionDelay(donPeerIDs []ragep2ptypes.PeerID, sharedSecret [16]byte, transmissionID string, tc TransmissionConfig) (map[p2ptypes.PeerID]time.Duration, error) {
```

Comment:
why is the sharedSecret passed in? presumably it's quite important that it's not some other 16 byte thing, but i can't tell why. can you add a comment please